pkg/platform/kube/logProxy/elastic/factory.go

```diff
@@ -66,31 +55,41 @@ func CreateLogProxy(logger logger.Logger, config *platformconfig.ElasticSearchCo
 }
 
 func getVersionFromSearchEngineWithRetries(
-	url string,
+	config *platformconfig.ElasticSearchConfig,
 	maxRetries int,
 	retryInterval time.Duration,
 	timeout time.Duration,
-) (versionInfoInstance *versionInfo, err error) {
-
+) (*versionInfo, error) {
 	client := &http.Client{Timeout: timeout}
+	var versionInfoInstance *versionInfo
+	var err error
 
 	for attempt := 1; attempt <= maxRetries; attempt++ {
-		versionInfoInstance, err = getVersionFromSearchEngine(client, url)
+		versionInfoInstance, err = getVersionFromSearchEngine(client, config)
 		if err == nil {
```

Comment:
Add a validation that `config.URL != ""` before starting the loop.
You only check the the config != nil before calling the factory, but it might not be nil and the URL can still be empty.